File: RentalManager.Infrastructure/DTO/RentalAgreementDTO.cs
namespace RentalManager.Infrastructure.DTO;

public class RentalAgreementDto
{
    public int Id { get; init; }
    public EmployeeDto Employee { get; init; } = null!;
    public bool IsActive { get; init; }
    public ClientDto Client { get; init; } = null!;
    public IEnumerable<RentalEquipmentDto> RentalEquipment { get; init; } = null!;
    public IEnumerable<PaymentDto> Payments { get; init; } = null!;
    public string? Comment { get; init; }
    public int Deposit { get; init; }
    public int? TransportFrom { get; init; }
    public int TransportTo { get; init; }
    public DateTime DateAdded { get; init; }
}

File: RentalManager.WebApp/src/references/codegen/API/models/RentalAgreementDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ClientDto } from './ClientDto';
import type { EmployeeDto } from './EmployeeDto';
import type { PaymentDto } from './PaymentDto';
import type { RentalEquipmentDto } from './RentalEquipmentDto';

export type RentalAgreementDto = {
    id?: number;
    employee?: EmployeeDto;
    isActive?: boolean;
    client?: ClientDto;
    rentalEquipment?: Array<RentalEquipmentDto> | null;
    payments?: Array<PaymentDto> | null;
    comment?: string | null;
    deposit?: number;
    transportFrom?: number | null;
    transportTo?: number;
    dateAdded?: string;
};
